demo/web/src/main/notebooks/B. Data science, ML, & AI/BStar The Learn Library- Real-Time AI.md

```diff
@@ -0,0 +1,188 @@
+
+- `rows`
+  - A set of rows in a table.  The actual set of rows is specified in the learn function call.
+- `columns`
+  - A set of columns in a table.  The actual set of columns is specified in the learn function call.
+- `order`
+  - The contiguity of the output array.  There are two possibilities:
+    - `MemoryLayout.COLUMN_MAJOR` or `MemoryLayout.FORTRAN`: The return array is stored in column-major order.
+    - `MemoryLayout.ROW_MAJOR` or `MemoryLayout.C`: The return array is stored in row-major order.
+  - Array contiguity is row major by default.  Array contiguity becomes a significant factor in performance when data becomes sufficiently large.
+- `dtype`
+  - The [NumPy dtype](https://numpy.org/doc/stable/reference/arrays.dtypes.html) of data in the return array.
+    - There exists a 1:1 mapping between supported [NumPy dtypes](https://numpy.org/doc/stable/reference/arrays.dtypes.html) and Java primitives.  Supported types are:
+      - Bytes, shorts, integers, longs, floats, and doubles
+
+The scatter function will return a single value from the output of the model function.  In this case, the output is a five-element-long 1D NumPy array.  A scatter function will _always_ take two inputs.  In the case that a model function returns a single value, the scatter function will look like this:
+
+```python
+def numpy_to_table(data, index):
+    return data
+```
+
+### The learn function call
+
+This is where everything comes together.  This function call is where everything that's already been set up gets used.  The learn function takes five inputs:
+
+- `table`
+  - The table that contains the data to process.  In the example, the table is `t`.
+- `model_func`
+  - The function that applies processing to data.  In the example, the function is `sum_columns`.
+- `inputs`
+  - The set of columns which will be used for data processing.  These must come in the form of one or more `learn.Input` objects.  In the example, the rows are `X` and `Y`.  Additionally, the gather function is specified here.  In the example, the gather function is `table_to_numpy_integer`.
+- `outputs`
+  - The set of outputs which will be seen as new columns in the output table.  These must come in the form of one or more `learn.Output` objects.  In the example, the output column is `Z`.  Additionally, the scatter function is specified here.  In the example, the scatter function is `numpy_to_table`.  Lastly, you can specify the Java primitive type of the output column.  This output must be a string, and correspond to Java primitive.  In this case, a primitive `int` is chosen as the output type.
+- `batch_size`
+  - The maximum number of rows in the table that will be gathered in a single iteration.  In the example, the batch size is the number of rows in `t`.
+
+## A static example
```

Comment:
The static and real-time examples are the fun part. You start with the simple example / syntax, but perhaps you could as easily break down the code below - doing it once; the step by step could come after observations. What do @hythloda and @jakemulf thinks?